### PR TITLE
Make all functions in this module testable on Travis CI

### PR DESCRIPTION
**Pull request by Jiu9Shen**
Since drush functions are not installed in travis ci, we have to make all drush functions optional.

### CODE REVIEW

File: api/vcf_filter.api.inc [hunk 1]
```diff
@@ -56,10 +56,13 @@ function vcf_filter_get_vcf_file($vcf_file_id) {
  * @return
  *   The full patch of the filtered VCF file.
  */
-function vcf_filter_filter_file($file_path, $filter_criteria) {
+function vcf_filter_filter_file($file_path, $filter_criteria, $use_4_unit_test = false) {
```
Comment (laceysanderson) [inline api/vcf_filter.api.inc]: I'm concerned this variable name is a little limiting. There may be other times when we want to turn drush off. I would have been tempted to change it to `$drush_available = TRUE` as a default but that will flip your if statements. Anyway, I like the verbosity but you should make the variable name more specific to drush and leave out the unit testing connection.

File: api/vcf_filter.api.inc [hunk 2]
```diff
@@ -56,10 +56,13 @@ function vcf_filter_get_vcf_file($vcf_file_id) {
...
   $new_file = variable_get('file_temporary_path') . '/' . uniqid();
 
-  drush_log('NOTE: We are not using the PL at this point to specify the genotypes.', 'warning');
+  if ($use_4_unit_test === false){
```
[the remark this answers is not in the document]
Comment (Jiu9Shen) [reply api/vcf_filter.api.inc]: I don't know about this one. Do you still want me to just print it to terminal or create a log file and print into file.

File: includes/vcf_filter.trpdownload.vcf.inc [hunk 3]
```diff
@@ -18,46 +18,58 @@
...
+ * @param $drush_use_limited
+ *   are drush function limited
  */
-function vcf_filter_vcf_generate_file($variables, $job_id = NULL) {
+function vcf_filter_vcf_generate_file($variables, $job_id = NULL, $drush_use_limited = false) {
 
+  /*for writting tests, should be removed after
+  print_r($variables);
+  print "\n";
+  */
```
Comment (laceysanderson) [inline includes/vcf_filter.trpdownload.vcf.inc]: A convention I use to ensure I can find things like this again is to use the `@debug` tag. For example,
```
// @debug print "Variables in vcf_filter_vcf_generate_file():\n".print_r($variables,TRUE)."\n";
```
Notice the use of `print_r($var,TRUE)` which tells it to print inline. Then as you are debugging you just click before the print and add a carriage return:
```
// @debug 
print "Variables in vcf_filter_vcf_generate_file():\n".print_r($variables,TRUE)."\n";
```
This way you can always grep @debug to find such statements in your code.

File: includes/vcf_filter.trpdownload.vcf.inc [hunk 4]
```diff
@@ -18,46 +18,58 @@
...
-
+  $filepath = $variables['fullpath'] . $variables['filename'];
+  print "File: ".$filepath."\n";
 
   // Get the VCF file details.
   $vcf_file = vcf_filter_get_vcf_file($variables['q']['vcf_file_id']);
-  drush_print("\nVCF File (".$vcf_file->name."): ".$vcf_file->file_path);
+  print "\nVCF File (".$vcf_file->name."): ".$vcf_file->file_path."\n";
+  //for writting tests, should be removed after
+  print_r($vcf_file);
```
Comment (laceysanderson) [inline includes/vcf_filter.trpdownload.vcf.inc]: Please comment out each of these when submitting PRs :-) See comment above for how to do so in such a way as to make it easy for continued testing.

File: tests/DrushUnitTest.php [hunk 5]
```diff
@@ -0,0 +1,137 @@
...
+        "description" => $faker->text,
+    );
+    drupal_write_record('vcf_files', $one_datafile);
+    $variables['q']['vcf_file_id'] = $one_datafile['vcf_file_id'];
+    return $variables;
+    //return array($variables, $datafile_fake);
+  }
+
+  /**
+   * Several tests for different formats.
```
Comment (laceysanderson) [inline tests/DrushUnitTest.php]: Drupal standards are to add @section to this docblock.
```
/**
 * @section
 * Short one-line heading
 *
 * Further description if needed...
 */
```

File: tests/DrushUnitTest.php [hunk 6]
```diff
@@ -0,0 +1,137 @@
...
+   * Tests must begin with the word "test".
+   * See https://phpunit.readthedocs.io/en/latest/ for more information.
+   * reasons to split into different test functions are:
+   *  1. each time test_file can be regenerated with unique id, and random vcftool parameters
+   *  2. maybe good for more specific and expanded tests for each format for next
+   */
+
+  // VCF format
+  public function testVCFFormat() {
```
Comment (laceysanderson) [inline tests/DrushUnitTest.php]: Each function dshould have it's own docblock. For example,
```
/**
 * Test VCF Format.
 *
 * Specifically testing vcf_filter_vcf_generate_file():
 *   - result file exists
 *   - result file is not empty
 */
public function testVCFFormat() {
```

File: tests/DrushUnitTest.php [hunk 7]
```diff
@@ -0,0 +1,137 @@
...
+    //import test file
+    $variables_test = $this -> generate_test_file();
+    print "\n\n\n\nStarting test for VCF format with: \n" . "Filename: " . $variables_test["filename"] . "\nFullpath: " . $variables_test["fullpath"] . "\nFile ID: " . $variables_test['q']['vcf_file_id'] . "\n\n";
+    $result_file_vcf = $variables_test['fullpath'].$variables_test['filename'];
+    //execute function and do tests
```
Comment (laceysanderson) [inline tests/DrushUnitTest.php]: Whitespace, especially empty lines is a valuable tool for making your code more readable ;-)

File: includes/vcf_filter.trpdownload.vcf.inc (hunk 3 above)
Comment (Jiu9Shen) [reply includes/vcf_filter.trpdownload.vcf.inc]: Really like this idea, very cool and smart! 
Thank you!

**Pull request merged**